graphblas/core/matrix.py

```diff
@@ -2805,6 +2805,124 @@ def power(self, n, op=semiring.plus_times):
             dtype=self.dtype,
         )
 
+    def setdiag(self, values, k=0, *, mask=None, accum=None, clear_missing=True, **opts):
+        """Set k'th diagonal with a Scalar, Vector, or array.
+
+        This is not a built-in GraphBLAS operation. It is implemented as a recipe.
+
+        Parameters
+        ----------
+        values : Vector or list or np.ndarray or scalar
+            New values to assign to the diagonal. The length of Vector and array
+            values must match the size of the diagonal being assigned to.
+        k : int, default=0
+            Which diagonal or off-diagonal to set. For example, set the elements
+            ``A[i, i+k] = values[i]``. The default, k=0, is the main diagonal.
+        mask : Mask, optional
+            Vector or Matrix Mask to control which diagonal elements to set.
+            If it is Matrix Mask, then only the diagonal is used as the mask.
+        accum : Monoid or BinaryOp, optional
+            Operator to use to combine existing diagonal values and new values.
+        clear_missing : bool, default=True
+            If True, missing elements in Vector values will result in the
+            corresponding diagonal elements to be removed. This parameter has
+            no effect if values is not a Vector or if accum operator is given.
+            The default is True to match normal behavior for assignments, but
+            it requires a call to ``select.offdiag``. If clear_missing is False,
+            then the recipe is approximately ``D = v.diag(k) ; self(D.S) << D``.
```

Comment:
I think `clear_missing` should be eliminated. It adds very little utility, adds mental complexity to the API, and the same effect can be achieved with the correctly chosen `accum` (i.e. clear_missing=False is the same as accum=second).

If someone really needed to accumulate diagonal values only where the incoming values are not missing, then they would use a mask. This requires a deeper understanding of how GraphBLAS works, yes, but it achieves the goal of making easy things simple and hard things possible.

Reply:
Oh, right you are! Removed. Much better :)